test/extended/cli/logs.go

```diff
@@ -0,0 +1,128 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"k8s.io/apimachinery/pkg/watch"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func readPodFixture(path string) (*corev1.Pod, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	content, err := kyaml.ToJSON(data)
+	if err != nil {
+		return nil, err
+	}
+	obj, err := runtime.Decode(scheme.Codecs.UniversalDecoder(corev1.SchemeGroupVersion), content)
+	if err != nil {
+		return nil, err
+	}
+	return obj.(*corev1.Pod), err
+}
+
+var _ = g.Describe("[cli] oc logs", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("cli-deployment", exutil.KubeConfigPath())
+
+	g.JustBeforeEach(func() {
+		// FIXME: remove this when https://github.com/openshift/origin/issues/20225 gets fixed
+		g.By("waiting for default service account")
+		err := exutil.WaitForServiceAccount(oc.KubeClient().CoreV1().ServiceAccounts(oc.Namespace()), "default")
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	var (
+		echoPodFixture = exutil.FixturePath("testdata", "cli", "echo-pod.yaml")
+	)
+
+	g.It("should get all logs with --follow if the pod is gonna be terminated right after", func() {
```

Comment:
this isnt really the same case.

previously, we start a deployment, wait for it to be running, then call oc logs --follow on the deployment, which i think results in attempting to get the logs for the deployer pod which is terminated in the interim so the logs command fails but it seems to me the deploy pod actually succeeded.

Reply:
I think there are more cases then one. But this looks like the cause of one flake I was looking into from the cri-o failures - https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/9717/pull-ci-openshift-openshift-ansible-e2e-gcp-crio/335/
```
/tmp/openshift/build-rpms/rpm/BUILD/origin-3.11.0/_output/local/go/src/github.com/openshift/origin/test/extended/deployments/deployments.go:577
Expected
    <string>: --> pre: Running hook pod ...
    test pre hook executed
    --> pre: Success
    --> Scaling custom-deployment-1 to 2
to contain substring
    <string>: --> Reached 50%
/tmp/openshift/build-rpms/rpm/BUILD/origin-3.11.0/_output/local/go/src/github.com/openshift/origin/test/extended/deployments/deployments.go:594
```
seems like the deployment completed so it should have the full log.

We had kind of similar issues with docker loosing the last line of logs so this test is warranted anyways and something I would like to see cri-o conform to. 

I am also working on other PRs to make those deployment tests in question fail with better errors and more logging to find out the cause of other flakes.